File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        tester = {}

        for i, num in enumerate(nums):
            if target - num in tester:
                index = tester[target - num]
                if index < i:
                    return [index, i]
                else:
                    return [i, index]
            else:
                tester[num] = i

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-1.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        vector<int> dp(nums.size(), 1);

        int maxResult = 1;
        for (int i = nums.size() - 1; i >= 0; i--) {
            for (int j = i + 1; j < nums.size(); j++) {
                if (nums[j] > nums[i]) {
                    dp[i] = max(dp[i], 1 + dp[j]);
                    maxResult = max(maxResult, dp[i]);
                }
            }
        }

        return maxResult;
    }
};
